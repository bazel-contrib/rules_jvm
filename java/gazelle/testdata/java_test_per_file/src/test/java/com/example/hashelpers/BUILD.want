load("@rules_java//java:defs.bzl", "java_library", "java_test")

java_library(
    name = "hashelpers",
    testonly = True,
    srcs = ["Helper.java"],
    _gazelle_imports = [
        "com.example.myproject.App", # TODO: per-file tests should have per-test imports, excluding this one
        "com.google.common.math.IntMath",
        "java.lang.Exception", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.MatcherAssert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.Matchers", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Assert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Test", # TODO: per-file tests should have per-test imports, excluding this one
    ],
    _java_packages = ["_TESTONLY!com.example.hashelpers"],
)

java_test(
    name = "AppTest",
    srcs = ["AppTest.java"],
    _gazelle_imports = [
        "_TESTONLY!com.example.hashelpers",
        "com.example.hashelpers",
        "com.example.myproject.App",
        "com.google.common.math.IntMath", # TODO: per-file tests should have per-test imports, excluding this one
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.Matchers", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Assert",
        "org.junit.Test",
    ],
    _java_packages = ["com.example.hashelpers"],
    test_class = "com.example.hashelpers.AppTest",
)

java_test(
    name = "OtherAppTest",
    srcs = ["OtherAppTest.java"],
    _gazelle_imports = [
        "_TESTONLY!com.example.hashelpers",
        "com.example.hashelpers",
        "com.example.myproject.App",
        "com.google.common.math.IntMath", # TODO: per-file tests should have per-test imports, excluding this one
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert",
        "org.hamcrest.Matchers",
        "org.junit.Assert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Test",
    ],
    _java_packages = ["com.example.hashelpers"],
    test_class = "com.example.hashelpers.OtherAppTest",
)
