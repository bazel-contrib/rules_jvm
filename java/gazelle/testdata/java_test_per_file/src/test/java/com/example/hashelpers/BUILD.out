load("@rules_java//java:defs.bzl", "java_library", "java_test")

java_library(
    name = "hashelpers",
    testonly = True,
    srcs = ["Helper.java"],
    deps = [
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)

java_test(
    name = "AppTest",
    srcs = ["AppTest.java"],
    test_class = "com.example.hashelpers.AppTest",
    deps = [
        ":hashelpers",
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)

java_test(
    name = "OtherAppTest",
    srcs = ["OtherAppTest.java"],
    test_class = "com.example.hashelpers.OtherAppTest",
    deps = [
        ":hashelpers",
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)
